### PR TITLE
Replace deprecated isSite() and isAdmin() calls

### PR DESCRIPTION
**Pull request by frankmayer**
As the title says, it replaces the deprecated isSite() and isAdmin() calls with the new ones

### CODE REVIEW

**Comment by frankmayer** [comment]
On it...

**Comment by infograf768** [comment]
hmm... why was that deprecated?

**Comment by Bakual** [comment]
> hmm... why was that deprecated?

https://github.com/joomla/joomla-cms/pull/8971

**Comment by frankmayer** [comment]
This passes for now, but only after I reverted the changes in 'libraries/cms/pagination/pagination.php'
Has anyone any idea why the replacements failed there?

**Comment by frankmayer** [comment]
hmm, I think I know why... checking it out

**Comment by andrepereiradasilva** [comment]
@frankmayer i think you missed some.

/libraries/joomla/document/renderer/html/modules.php:37
/libraries/joomla/cache/cache.php:702
/libraries/joomla/cache/storage/redis.php:109
/libraries/joomla/cache/storage/redis.php:119
/libraries/joomla/cache/storage/redis.php:13
/libraries/joomla/cache/storage/redis.php:149
/libraries/joomla/form/field.php:1085

(also for consistency IMHO the same should be done in weblinks repository https://github.com/joomla-extensions/weblinks after)

**Comment by frankmayer** [comment]
@andrepereiradasilva are these not part of the platform? Shouldn't they be changed over there?

**Comment by frankmayer** [comment]
@wilsonge The problem was with mocking and I fixed it.
However, I am not able to understand what you meant with your lines. It's probably my logic failing me :smile: , but could you elaborate?

**Comment by andrepereiradasilva** [comment]
> @andrepereiradasilva are these not part of the platform? Shouldn't they be changed over there?

no. afaik there is no plataform repository. there is a framework repository if that is what you mean, but those are in the vendor dir (see https://github.com/joomla/joomla-cms/tree/staging/libraries/vendor/joomla and https://github.com/joomla-framework/)

**Comment by frankmayer** [comment]
@andrepereiradasilva Oh, ok, I confused them. Thanks for clarifying. Will change those ones, too. Already made a PR to weblinks as you suggested.

**Comment by wilsonge** [comment]
Everything in libraries/joomla, libraries/cms and libraries/legacy should be updated too :)

**Comment by frankmayer** [comment]
@wilsonge will check those dirs, also. :)

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 97ad36bdfb2a7a4e66b6ef1d6b22977be23d98cf<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13156">issues.joomla.org/tracker/joomla-cms/13156</a>.</sub>

**Comment by frankmayer** [comment]
Conflicts resolved...

File: components/com_config/controller/helper.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function parseController($app)
 		// Some special handling for com_config administrator
 		$option = $app->input->get('option');
 
-		if ($option === 'com_config' && $app->isAdmin())
+		if ($app->isClient('administrator') && $option == 'com_config')
```
Comment (andrepereiradasilva) [inline components/com_config/controller/helper.php]: don't temove the strict comparation here

File: plugins/system/debug/debug.php [hunk 2]
```diff
@@ -2079,7 +2079,7 @@ protected function prettyPrintJSON($json = '')
 	protected function writeToFile()
 	{
 		$app    = JFactory::getApplication();
-		$domain = $app->isSite() ? 'site' : 'admin';
+		$domain = ($app->isClient('site')) ? 'site' : 'admin';
```
Comment (andrepereiradasilva) [inline plugins/system/debug/debug.php]: don't add the parenthisis here

File: components/com_config/controller/helper.php (hunk 1 above)
Comment (frankmayer) [inline components/com_config/controller/helper.php]: Conflict resolution... sorry

**Review by andrepereiradasilva — APPROVED** [review]

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on dad2bc3777a23b1bbbab5f2d18918654dd69a9a8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13156">issues.joomla.org/tracker/joomla-cms/13156</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on dad2bc3777a23b1bbbab5f2d18918654dd69a9a8<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13156">issues.joomla.org/tracker/joomla-cms/13156</a>.</sub>

**Comment by frankmayer** [comment]
Conflicts resolved...

File: components/com_config/controller/helper.php [hunk 3]
```diff
@@ -90,7 +90,7 @@ public function parseController($app)
 		// Some special handling for com_config administrator
 		$option = $app->input->get('option');
 
-		if ($option === 'com_config' && $app->isAdmin())
+		if ($app->isClient('administrator') && $option === 'com_config')
```
Comment (andrepereiradasilva) [inline components/com_config/controller/helper.php]: Why change the order here?

Comment (frankmayer) [inline components/com_config/controller/helper.php]: by accident... during conflict resolving I guess. Thanks for catching. :+1:

**Review by andrepereiradasilva — APPROVED** [review]

**Pull request merged**